### PR TITLE
Add transaction details dropdown

### PR DESCRIPTION
**Pull request by jackburrus**
This PR adds the transaction details from the figma design. It is feature flagged.

https://github.com/user-attachments/assets/3883250f-5dc2-4291-900b-254d7873e283

### CODE REVIEW

File: apps/hyperdrive-trading/src/ui/hyperdrive/longs/OpenLongPreview/OpenLongPreview.tsx [hunk 1]
```diff
@@ -62,7 +65,97 @@ export function OpenLongPreview({
...
+    useFeatureFlag("new-open-long-form");
+  if (isNewOpenLongFormEnabled) {
+    return (
+      <div className="flex flex-col gap-3.5 px-2">
+        <CollapseSection
+          heading={
+            <div className="flex w-full items-center justify-between text-neutral-content">
+              <p>Transaction Details</p>
+              <div className="flex items-center gap-1">
+                <ClockIcon className="size-5 text-[#4E6A77]" />
```
Comment (jackburrus) [inline apps/hyperdrive-trading/src/ui/hyperdrive/longs/OpenLongPreview/OpenLongPreview.tsx:79]: This was the color used in Figma but I noticed we don't have it in our config. It looks like a useful shade to be paired with text-content-neutral. Let me know if you'd like me to add it.

File: apps/hyperdrive-trading/src/ui/base/components/CollapseSection/CollapseSection.tsx [hunk 2]
```diff
@@ -1,23 +1,16 @@
-import { ChevronDownIcon } from "@heroicons/react/20/solid";
 import { PropsWithChildren, ReactElement } from "react";
 
 export function CollapseSection({
   heading,
   children,
 }: PropsWithChildren<{
-  heading: string;
+  heading: ReactElement | string;
```
Comment (DannyDelott) [inline apps/hyperdrive-trading/src/ui/base/components/CollapseSection/CollapseSection.tsx:7]: this should just be `ReactNode`, which includes`string`

File: apps/hyperdrive-trading/src/ui/hyperdrive/TransactionView.tsx [hunk 3]
```diff
@@ -30,13 +33,18 @@ export function TransactionView({
       </div>
       <div className="flex flex-col gap-8">
         {primaryStats}
-        <div>
-          <h6 className="mb-4">Preview transaction</h6>
-          {transactionPreview}
-        </div>
+        {/* If the new open long form is not enabled, show the transaction preview above the action button */}
```
Comment (DannyDelott) [inline apps/hyperdrive-trading/src/ui/hyperdrive/TransactionView.tsx:36]: nice comments 🙏

File: apps/hyperdrive-trading/src/ui/hyperdrive/longs/OpenLongPreview/OpenLongPreview.tsx (hunk 1 above)
Comment (DannyDelott) [reply apps/hyperdrive-trading/src/ui/hyperdrive/longs/OpenLongPreview/OpenLongPreview.tsx:79]: Yeah it's a lower luminance `text-neutral-content` (IIRC, I played with the HSL colors): 
image

I like what you're saying about adding it as a proper tailwind color. How about making it the `gray-500` value?

image

**Review by DannyDelott — APPROVED** [review]
Nice!

File: apps/hyperdrive-trading/src/ui/base/components/CollapseSection/CollapseSection.tsx (hunk 2 above)
Comment (jackburrus) [reply apps/hyperdrive-trading/src/ui/base/components/CollapseSection/CollapseSection.tsx:7]: Interesting, TIL

**Pull request merged**